tests/lib.rs

```diff
@@ -748,3 +748,18 @@ fn get_one() {
 
     assert_match!(r.get_one(&x.0).as_deref(), Some(('x', 42)));
 }
+
+#[test]
+fn insert_remove() {
+    let x = 'x';
+
+    let (r, mut w) = evmap::new();
+
+    w.insert(x, x);
+    w.refresh();
+
+    w.remove(x, x);
+    w.refresh();
+
+    assert!(r.is_empty());
```

Comment:
I was copying the [quicktests from `indexmap`](https://github.com/bluss/indexmap/blob/master/tests/quick.rs#L67) and the `insert_remove` test failed. I made a small version here and this assertion fails. Unfortunately ... I can't tell if it's supposed to pass haha. If it is, I can work on getting this test to pass. If it's working correctly, then I can adjust the test to document that!

Reply:
Hmm, that seems weird. Are you calling `remove`, `clear`, or `empty`? They all have slightly different semantics in evmap. Also, as always with evmap, keep in mind that nothing is visible to the reader until you've done a `refresh` on the `WriteHandle`!

Reply:
Good questions. Just `remove` for this test. And I do call `refresh` and see that the key no longer as associated values. But since the map still has a key (`remove` doesn’t remove keys) it isn’t considered empty and so neither is the overall `evmap` (I think).

It looks like `empty` would work because that calls `remove` on the inner map instead of the value of the key. But I guess I’m asking, “After calling `remove` on the last key-value pair, should the map be empty? Or should it retain the key for some purpose?” Once we know that, then we know if we need to update the test or update the code! :-)

Reply:
So, `remove` removed _one_ value from a value-bag for a given key. Its semantics are (at least at the moment, and should probably remain) that it does not erase the key even when the last value is removed. That's what `empty` is for. Arguably the names could be better though :p

Reply:
Haha I can throw up a quick PR to rename `remove` to `remove_value` and then a follow-up PR to rename `empty` to `remove` to make it match a regular map more closely. Or not!

Reply:
Hmm... That would be a backwards-incompatible change, which is a little sad. Maybe mark `empty` and `remove` as deprecated (with `#[deprecated]`), and add `remove_value` and `remove_entry`?

Reply:
Great idea! 👍